### ISSUE TITLE
Still active

### ISSUE THREAD

**Issue by Fusion** [sonicnkt/glauth-ui#16]
Hi Nils,

I would like to check whether you are still actively working on this project. This is a very nice addition to the GLAuth ecosystem so I hope you are! ;)

**Comment by sonicnkt** [comment]
Hi Chris,
didnt have much time those last months but im still actively using the glauth and the ui at work and at home and plan to improve it further. I have been doing most of the work when i did not have much to do at work but this did ramp up in the last months again.
Anyway it should be getting more quite now at work again.

**Comment by sonicnkt** [comment]
So finally got into this again. 
The current main branch now finally supports glauth 2.x but no other updates yet. It took me quite some time to get back into this but i really want to do more now  and support all the great new features in glauth 2.1.

**Comment by sonicnkt** [comment]
Just wanted to inform anyone that im currently in the process of getting into glauth-ui dev again :) (this time i hope for real)

I haven't done anything with flask in a while and could not find the time to get into it again. 
I just got little job tho developing a (very) small web app and put this together with flask which trigged my interest again :D 

I will probably merge the latest PRs (THANKS!) in the next days and then work on new/missing features.

Features/Stuff i want to add/improve (in order):
 - Add support for behvaiors and user capibilities
 - Improve bcrypt password support (migration from sha to bcrypt)
 - Add Documentation for the basic forward auth implementation.
 - Add missing user attributes (optional)
 - Add support for optional custom and preconfigured attirbutes
 - Add glauth statistics to admin ui (using rest api)
 - Add support for multifactor authentication
 - Add support for custom configs (
 
I am also thinking of bundling the glauth binary with the ui, this would allow me to show realtime logs in the admin interface and also more server controls like stop/start/restart.

**Comment by kerberg** [comment]
Hello Nils,
Thanks for sharing this development. 
I wanted ask if its possible to make a front end website based user interface that would allow people to register and request authentication? Would love to learn more about how (or if!?) this could be developed.
Thanks again!
Ethan

**Comment by sonicnkt** [comment]
> I wanted ask if its possible to make a front end website based user interface that would allow people to register and request authentication? Would love to learn more about how (or if!?) this could be developed.

This would be really simple. I created this app while going through the [mega flask tutorial by miguel grinberg](https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-i-hello-world).  You would only have to create a new route that allows registration, i didn't add this on purpose as i only wanted users created by the admin. 
I would not be against it tho if someone creates a PR, only condition would be that its disabled by default and configurable through an env variable (and/or a setting in the admin panel). 

if no one creates a pr i  could add it myself ,but with low priority as don't have much time right now. 
The other app i am currently still working on took longer than expected....

**Comment by helmut72** [comment]
Time goes by ... still active?

**Comment by ericdai** [comment]
Did it also support openldap project，can be used to manager openLdap ？

**Comment by johndo100** [comment]
> Did it also support openldap project，can be used to manager openLdap ？

Nope, it's for glauth only. I looked into the code, it change data in config file and databases instead of make a ldap request so It can not support other ldap server than glauth.

I'm trying this UI on my machine now. It's good btw.

**Comment by Fusion** [comment]
> I'm trying this UI on my machine now. It's good btw.

I like this project a lot 😄

**Comment by johndo100** [comment]
> > I'm trying this UI on my machine now. It's good btw.
> 
> 
> 
> I like this project a lot 😄 
> 
> 

It's good, the UI and I like it.
But I think it's better if we can do more dynamic config by this UI.
It seems create static files and overwrite for each time I save.
So I will lost my manual config.

**Comment by Fusion** [comment]
There is a not-so-well-documented feature, in GLAuth, (because it's beta quality!) that lets you save your configuration in a folder instead, and stitches these config files together.
I imagine, if poor sonicnkt found the time to support this, it would address your concern.
Of course, a fair reply to that would be that perhaps support for configuration folders should stop being a beta feature.
